plugins/huggingface/modelgauge/suts/huggingface_chat_completion.py

```diff
@@ -79,29 +85,38 @@ def translate_text_prompt(self, prompt: TextPrompt) -> HuggingFaceChatCompletion
             **prompt.options.model_dump(),
         )
 
-    def evaluate(self, request: HuggingFaceChatCompletionRequest) -> ChatCompletionOutput:
+    def evaluate(self, request: HuggingFaceChatCompletionRequest) -> HuggingFaceChatCompletionOutput:
         if self.client is None:
             self._create_client()
 
         request_dict = request.model_dump(exclude_none=True)
-        return self.client.chat_completion(**request_dict)  # type: ignore
+        response = self.client.chat_completion(**request_dict)  # type: ignore
+        # Convert to cacheable pydantic object.
+        return HuggingFaceChatCompletionOutput(
+            choices=[asdict(choice) for choice in response.choices],
+            created=response.created,
+            id=response.id,
+            model=response.model,
+            system_fingerprint=response.system_fingerprint,
+            usage=asdict(response.usage),
+        )
 
     def translate_response(
-        self, request: HuggingFaceChatCompletionRequest, response: ChatCompletionOutput
+        self, request: HuggingFaceChatCompletionRequest, response: HuggingFaceChatCompletionOutput
     ) -> SUTResponse:
         completions = []
         for choice in response.choices:
-            text = choice.message.content
+            text = choice["message"]["content"]
```

Comment:
You may want to try the [benedict library](https://pypi.org/project/python-benedict/). It wraps `dict` and gives you convenient functionality like accessing elements via dot notation or dict notation interchangeably (`choice["message"]` or `choice.message`).